circuit/types/field/src/div_flagged.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+
+impl<E: Environment> DivFlagged<Field<E>> for Field<E> {
+    type Output = (Field<E>, Boolean<E>);
+
+    /// Performs flagged division of two field elements.
+    #[doc(hidden)]
+    fn div_flagged(&self, other: &Field<E>) -> Self::Output {
+        match (self.is_constant(), other.is_constant()) {
+            // If `other` is a constant and zero, return the zero element and set the flag.
+            (_, true) if other.eject_value().is_zero() => (Field::zero(), Boolean::constant(true)),
```

Comment:
I take it this is being done because any nontrivial ring/field's zero element doesn't have a multiplicative inverse? Is it possible to document here with a sentence or to why this is implemented - namely what the consequences of allowing a finite field arithmetic violation like this to happen without a flag?

Reply:
We're doing this to ensure that the division operation is totally defined.
This will be used to create an instruction called `DivFlagged`, likely `div.f`, which is used to implement non-strict semantics in Leo.
In the standard field division circuit, a division be zero results in a failed execution.
I believe that it will be sufficient to document these new instructions in our opcode reference.